File: common/src/main/java/com/nmh/commerce/support/error/CustomException.java
package com.nmh.commerce.support.error;

import lombok.Getter;

@Getter
public abstract class CustomException extends RuntimeException {

    protected final ErrorType errorType;

    protected final Object data;

    public CustomException(ErrorType errorType) {
        super(errorType.getMessage());
        this.errorType = errorType;
        this.data = null;
    }

    public CustomException(ErrorType errorType, Object data) {
        super(errorType.getMessage());
        this.errorType = errorType;
        this.data = data;
    }
}

File: common/src/main/kotlin/com/nmh/commerce/support/error/CustomException.kt
package com.nmh.commerce.support.error

abstract class CustomException : RuntimeException {
    val errorType: ErrorType

    val data: Any?

    constructor(errorType: ErrorType) : super(errorType.message) {
        this.errorType = errorType
        this.data = null
    }

    constructor(errorType: ErrorType, data: Any?) : super(errorType.message) {
        this.errorType = errorType
        this.data = data
    }
}
